dashjoin-rdf4j/src/main/java/org/dashjoin/service/RDF4J.java

```diff
@@ -499,27 +512,74 @@ void add(Table table, Map<String, Object> res, IRI predicate, Value value, boole
   }
 
   @Override
-  public List<Map<String, Object>> queryGraph(QueryMeta info, Map<String, Object> arguments)
-      throws Exception {
-    List<Map<String, Object>> res = queryInternal(info, arguments, true);
-    return res;
-  }
+  @SuppressWarnings("unchecked")
+	public List<Map<String, Object>> queryGraph(QueryMeta info, Map<String, Object> arguments) throws Exception {
+		List<Map<String, Object>> res =  new ArrayList<>();
+		String[] queryParts = info.query.split(IntelligentGraphConstants.PATH_QL_REGEX);
+		switch (queryParts[0]) {
+			case "getFact":
+			case "getFacts":
+				break;
+			case "getPath":
+			case "getPaths":
+				IRI predicate = iri(PATHQL.NAMESPACE + URLEncoder.encode(info.query, StandardCharsets.UTF_8.toString()));
+				IRI subject = (IRI) info.arguments.get("subject");
+				Value object = (IRI) info.arguments.get("object");
+				PathResults pathsIterator = null;
+				try (RepositoryConnection con = getConnection()) {
+					RepositoryResult<Statement> resultsIterator = con.getStatements(subject, predicate, object);
+					Thing subjectThing = Thing.create(null, subject, null);
+					pathsIterator = new PathResults(resultsIterator, subjectThing, null, null);
+	
+					while (pathsIterator.hasNext()) {
+						Path path = pathsIterator.next();
+					    PathSteps pathSteps= new PathSteps(info.database , path);
+					    @SuppressWarnings("rawtypes")
+						Map row = new LinkedHashMap(pathSteps);
+					    res.add(row);
+					}
+				}catch(Exception e){
+					System.out.println(e.getMessage());
+				}finally {
+					if(pathsIterator!=null) pathsIterator.close();
+				}
+				break;
+			default:
+				res = queryInternal(info, arguments, true);
+
+		}
+		return res;
+	}
 
   @Override
   @SuppressWarnings("unchecked")
   public Map<String, Object> connectAndCollectMetadata() throws Exception {
-    if ("memory".equals(mode))
-      _cp = new SailRepository(new MemoryStore());
-    if ("local".equals(mode))
-      _cp = new SailRepository(new NativeStore(new File(folder)));
-    if ("sesame".equals(mode)) {
-      _cp = new HTTPRepository(endpoint);
-      ((HTTPRepository) _cp).setUsernameAndPassword(username, password);
-    }
-    if ("client".equals(mode)) {
-      _cp = new SPARQLRepository(endpoint);
-      ((SPARQLRepository) _cp).setUsernameAndPassword(username, password);
-    }
+	  IntelligentGraphConfig intelligentGraphConfig = null;

```

Comment:
we use the default eclipse java formatter. Would be good to keep the style consistent

Reply:
Not sure I understand, I *think* I use the default formatter as well.